### ISSUE TITLE
Problem with pip installation

### ISSUE THREAD

**Issue by marcorusc** [saezlab/pypath#192]
**Describe the bug**
Hello!
During pip installation (pip install git+https://github.com/saezlab/pypath.git) it raises the following error: 

pip._vendor.pkg_resources.RequirementParseError: Parse error at "'extra =='": Expected stringEnd

It could totally be me, but I have no clue on how to fix this...

**To Reproduce**
- Open the terminal (linux)
- pip install git+https://github.com/saezlab/pypath.git

**Expected behavior**
Install pypath

**Traceback**
__________________________________________________________________________________________________________
Collecting git+https://github.com/saezlab/pypath.git
  Cloning https://github.com/saezlab/pypath.git to /tmp/pip-req-build-g64gz48f
  Running command git clone --filter=blob:none -q https://github.com/saezlab/pypath.git /tmp/pip-req-build-g64gz48f
  Resolved https://github.com/saezlab/pypath.git to commit ccbfe000f27144b8bcfb889cdf90ed868bd81704
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
ERROR: Exception:
Traceback (most recent call last):
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3021, in _dep_map
    return self.__dep_map
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2815, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/packaging/requirements.py", line 102, in __init__
    req = REQUIREMENT.parseString(requirement_string)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pyparsing.py", line 1955, in parseString
    raise exc
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pyparsing.py", line 3814, in parseImpl
    raise ParseException(instring, loc, self.errmsg, self)
pip._vendor.pyparsing.ParseException: Expected stringEnd, found 'e'  (at char 27), (line:1, col:28)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3101, in __init__
    super(Requirement, self).__init__(requirement_string)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/packaging/requirements.py", line 104, in __init__
    raise InvalidRequirement(
pip._vendor.packaging.requirements.InvalidRequirement: Parse error at "'extra =='": Expected stringEnd

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/cli/base_command.py", line 164, in exc_logging_wrapper
    status = run_func(*args)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/cli/req_command.py", line 205, in wrapper
    return func(self, options, args)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/commands/install.py", line 338, in run
    requirement_set = resolver.resolve(
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 92, in resolve
    result = self._result = resolver.resolve(
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/resolvelib/resolvers.py", line 482, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/resolvelib/resolvers.py", line 374, in resolve
    failure_causes = self._attempt_to_pin_criterion(name)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/resolvelib/resolvers.py", line 214, in _attempt_to_pin_criterion
    criteria = self._get_updated_criteria(candidate)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/resolvelib/resolvers.py", line 204, in _get_updated_criteria
    for requirement in self._p.get_dependencies(candidate=candidate):
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/resolution/resolvelib/provider.py", line 204, in get_dependencies
    return [r for r in candidate.iter_dependencies(with_requires) if r is not None]
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/resolution/resolvelib/provider.py", line 204, in <listcomp>
    return [r for r in candidate.iter_dependencies(with_requires) if r is not None]
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 236, in iter_dependencies
    requires = self.dist.iter_dependencies() if with_requires else ()
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_internal/metadata/pkg_resources.py", line 91, in iter_dependencies
    return self._dist.requires(extras)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2736, in requires
    dm = self._dep_map
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3023, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3033, in _compute_dependencies
    reqs.extend(parse_requirements(req))
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3094, in parse_requirements
    yield Requirement(line)
  File "/home/marco/anaconda3/envs/pypath_env/lib/python3.10/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3103, in __init__
    raise RequirementParseError(str(e))
pip._vendor.pkg_resources.RequirementParseError: Parse error at "'extra =='": Expected stringEnd

____________________________________________________________________________________________________________

**Log file**
no log file

**Desktop (please complete the following information):**
 - OS: [Ubuntu 22.04]
 - Python version: [ 3.8]
 - Version or commit hash [last commit]

**Additional context**
Searching around, it looks like there should be a problem somewhere with an url missing a space at the end. I have found a similar problem here:
https://github.com/python-poetry/poetry/issues/2824
At first I thought it was a problem with pip, so I tried to update it, still nothing solved...
In the last released of pypath there is problem with uniprot.py, which should be solved in one of the last commit if I have understood properly.

**Comment by chrarnold** [comment]
I receive exactly the same, on an Ubuntu system, with Python 3.10. Looks like a real issue, what to do?

**Comment by deeenes** [comment]
Hi @marcorusc & @chrarnold ,

Apologies for the late reply. This issue should be fixed now. It was caused by that we added a development dependency as local git submodule, which meant a directory that obviously doesn't exist in the package. Unfortunately it looks like [poetry](https://python-poetry.org/) is not mature enough yet to handle such a dependency. Hence now I completely removed it.

Best,

Denes
